src/envs/pybullet_env.py

```diff
@@ -329,37 +329,81 @@ def _create_move_end_effector_to_pose_option(
                     state, objects, params)
-            # Run IK to determine the target joint positions.
+            target_joints = self._pybullet_robot.run_inverse_kinematics(
+                target, validate=True)
+            # Soon, we will want to do motion planning and check collisions
+            # here, but for now, linear interpolation in joint space is ok.
+            # Use the difference between the current and target poses to
+            # determine how many steps to take so that we conform to the
+            # max velocity constraint.
             ee_delta = np.subtract(target, current)
-            # Reduce the target to conform to the max velocity constraint.
             ee_norm = np.linalg.norm(ee_delta)  # type: ignore
-            if ee_norm > self._max_vel_norm:
-                ee_delta = ee_delta * self._max_vel_norm / ee_norm
-            ee_action = np.add(current, ee_delta)
-            # We assume that the robot is already close enough to the target
-            # position that IK will succeed with one call, so validate is False.
-            # Furthermore, updating the state of the robot during simulation,
-            # which validate=True would do, is discouraged by PyBullet.
-            joint_state = self._pybullet_robot.run_inverse_kinematics(
-                (ee_action[0], ee_action[1], ee_action[2]), validate=False)
+            # Special case: we're already at the target. We should terminate
+            # successfully in one step.
+            if ee_norm == 0.0:
+                num_steps = 1
+            else:
+                num_steps = int(np.ceil(ee_norm / self._max_vel_norm))
+            plan = np.linspace(current_joints, target_joints, num_steps)
+            memory["waypoints"] = plan.tolist()
+            memory["finger_status"] = finger_status
+            # Always initiable.
+            return True
+
+        def _policy(state: State, memory: Dict, objects: Sequence[Object],
+                    params: Array) -> Action:
+            del objects, params  # unused
+            assert "waypoints" in memory and "finger_status" in memory, \
+                "initiable() must be called before policy()"
+            # Extract the current joint state.
+            current_joints = cast(_PyBulletState, state).joint_state
+            # Extract the current finger state from the simulator state.
+            finger_state = self._get_finger_state(state)
+            finger_status = memory["finger_status"]
+            # Determine if we have reached the next waypoint in the plan.
+            # The only way that the plan would be empty is if we were
+            # already close enough to the final waypoint, i.e. the target,
+            # in which case the option should have terminated and the
+            # policy should no longer be getting called.
+            waypoint = memory["waypoints"][0]
+            # The finger states should not be used in the distance computation
```

Comment:
how are they not being used? they're set to be part of the waypoints list:
```
waypoint[self._pybullet_robot.left_finger_joint_idx] = finger_state
waypoint[self._pybullet_robot.right_finger_joint_idx] = finger_state
```
and then used here: 
```
squared_dist = np.sum(np.square(np.subtract(current_joints, waypoint)))
```